### DISCUSSION TITLE
Got the same molecular 3D coordinate for two chiral molecules

### DISCUSSION THREAD

**Discussion by zwang1995** [rdkit/rdkit#3951] [Development]
I faced a problem in getting 3D coordinates for two chiral molecules.
Apparently, the following two SMILES have different 3D coordinates, but it gives the same one.

```
m1 = Chem.MolFromSmiles('C[C@H](CCN)O')
m2 = Chem.MolFromSmiles('C[C@@H](CCN)O')
AllChem.EmbedMolecule(m1)
AllChem.EmbedMolecule(m2)
print(Chem.MolToMolBlock(m1))
print(Chem.MolToMolBlock(m2))
```

Different 3D coordinates can be generated if I deploy **AllChem.AddHs()** before get the coordinates. 
But I don't want to add Hs for them.
So, is there a solution to generate different 3D coordinates for the above two molecules without **AllChem.AddHs()**.

**Comment by greglandrum** [comment]
You cannot expect to get reasonable 3D coordinates if you don't add Hs.

If you don't want to see the Hs after you generate the coordinates you can use the pattern:
```
n [4]: mh = Chem.AddHs(m)

In [5]: AllChem.EmbedMolecule(mh)
Out[5]: 0

In [6]: m = Chem.RemoveHs(mh)
```

**Reply by zwang1995** [reply]
I got it. Thank you!
